### PR TITLE
Compile with newer openal-soft

### PR DESCRIPTION
**Pull request by Cloudef**
See the title.

### CODE REVIEW

File: src/eventthread.h [hunk 1]
```diff
@@ -37,7 +37,7 @@
 #include <stdint.h>
 
 struct RGSSThreadData;
-typedef struct ALCdevice_struct ALCdevice;
+typedef struct ALCdevice ALCdevice;
```
Comment (carstene1ns) [inline src/eventthread.h]: This should also work with both versions: `struct ALCdevice;`
Can you try this?

Comment (Cloudef) [reply src/eventthread.h]: Yeah, I can try this.

Comment (Cloudef) [reply src/eventthread.h]: Sadly doesn't work. We can either do preprocessor #if or include alc.h in the header.
https://github.com/kcat/openal-soft/commit/399dfca1e3352050a5289be6bcd7b7dce353ccf8
openal-soft doesn't seem to consider using the struct as public api, so including alc.h seems to be only sane way. Sadly they typedef their thing too so we can't really forward declare. (I'm not really fond of typedef)

**Comment by Cloudef** [comment]
Updated to include alc.h in the header and removed the typedef.

**Comment by carstene1ns** [comment]
@Ancurio We should merge this (build is broken on updated systems and fix should work on older as well)

**Comment by Cloudef** [comment]
Hopefully they are still alive

**Comment by Ancurio** [comment]
Sorry for the holdup, I'll get to this soon.

**Pull request merged**